src/lib/party.ts

```diff
@@ -927,42 +932,29 @@ class Party implements Types.Party {
     return { body, authorization: Ledger.dummySignature() };
   }
 
-  static createUnsigned(publicKey: PublicKey) {
-    let party = Party.defaultParty(publicKey);
-    Mina.currentTransaction()?.parties.push(party);
+  static create(publicKey: PublicKey, tokenId?: Field) {
+    let party = Party.defaultParty(publicKey, tokenId);
+    if (smartContractContext.has()) {
+      makeChildParty(smartContractContext.get().this.self, party);
```

Comment:
apart from changing the (IMO, confusing) `createUnsigned` name to `create`, I also now automatically create a child party (instead of a top level party) when inside a zkapp. This is because typically when a zkapp creates a party, it wants to make statements about it. If it isn't a child party, those statements would not go in the proof, which I think is a huge security footgun